rest-service/manager_rest/rest/rest_utils.py

```diff
@@ -790,15 +812,40 @@ def propagate_deployment_statuses(self, source):
                     total_srv_status = _source.compare_between_statuses(
                         total_srv_status, _sub_srv_status
                     )
+            _update_deployment_status(v, total_srv_status, total_env_status)
 
-            if total_srv_status or total_env_status:
-                target = self.sm.get(models.Deployment, v, locking=True)
-                if total_env_status:
-                    target.sub_environments_status = total_env_status
-                if total_srv_status:
-                    target.sub_services_status = total_srv_status
-                target.deployment_status = target.evaluate_deployment_status()
-                self.sm.update(target)
+    def retrieve_and_display_dependencies(self, target):
+        self.create_dependencies_graph()
+        queue = [target.id]
+        visited = defaultdict(bool)
+        dependencies = []
+        while queue:
+            v = queue.pop()
```

Comment:
I note that by doing this we can get into an infinite loop if there are recursive dependencies.

Do we guard elsewhere against that? If not, maybe guard against it here by keeping a separate list of visited nodes and just continuing if you hit a node you've already visited.

Reply:
@geokala we already have a visited nodes and I add an extra check after pop from the queue if that node is already visited to continue otherwise mark it as visited